src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs

```diff
@@ -37,36 +49,69 @@ public IndexQueryServerSide(string query, BlittableJsonReaderObject queryParamet
         }
 
         public static IndexQueryServerSide Create(
+            HttpContext httpContext,
             BlittableJsonReaderObject json,
             QueryMetadataCache cache,
+            RequestTimeTracker tracker,
             QueryType queryType = QueryType.Select)
         {
             var result = JsonDeserializationServer.IndexQuery(json);
+            if (tracker != null)
+                tracker.Query = result.Query;
 
             if (result.PageSize == 0 && json.TryGet(nameof(PageSize), out int _) == false)
                 result.PageSize = int.MaxValue;
 
             if (string.IsNullOrWhiteSpace(result.Query))
-                throw new InvalidOperationException($"Index query does not contain '{nameof(Query)}' field.");
+            {
+                var errorMessage = $"Index query does not contain '{nameof(Query)}' field.";
+
+                if (TrafficWatchManager.HasRegisteredClients)
+                    AddStringToHttpContext(httpContext, errorMessage, TrafficWatchChangeType.Queries);
```

Comment:
Why not simply wrap the whole Query operation in a `try/catch` ?

Reply:
done